fundingmanager.go

```diff
@@ -840,109 +838,125 @@ func (f *fundingManager) reservationCoordinator() {
 
 	chanID := lnwire.NewChanIDFromOutPoint(&channel.FundingOutpoint)
-	fndgLog.Debugf("channel (%v) with opening state %v found",
-		chanID, channelState)
-
-	// If we did find the channel in the opening state database, we have
-	// seen the funding transaction being confirmed, but we did not finish
-	// the rest of the setup procedure before we shut down. We handle the
-	// remaining steps of this setup by continuing the procedure where we
-	// left off.
+	fndgLog.Debugf("Channel(%v) with ShortChanID %v has opening state %v.",
+		chanID, shortChanID, channelState)
+
 	switch channelState {
 
 	// The funding transaction was confirmed, but we did not successfully
 	// send the fundingLocked message to the peer, so let's do that now.
 	case markedOpen:
-		f.wg.Add(1)
-		go func(dbChan *channeldb.OpenChannel) {
-			defer f.wg.Done()
+		peerChan := make(chan lnpeer.Peer, 1)
 
-			peerChan := make(chan lnpeer.Peer, 1)
+		var peerKey [33]byte
+		copy(peerKey[:], channel.IdentityPub.SerializeCompressed())
 
-			var peerKey [33]byte
-			copy(peerKey[:], dbChan.IdentityPub.SerializeCompressed())
+		f.cfg.NotifyWhenOnline(peerKey, peerChan)
 
-			f.cfg.NotifyWhenOnline(peerKey, peerChan)
+		var peer lnpeer.Peer
+		select {
+		case peer = <-peerChan:
+		case <-f.quit:
+			return ErrFundingManagerShuttingDown
+		}
 
-			var peer lnpeer.Peer
-			select {
-			case peer = <-peerChan:
-			case <-f.quit:
-				return
-			}
-			err := f.handleFundingConfirmation(
-				peer, dbChan, shortChanID,
-			)
-			if err != nil {
-				fndgLog.Errorf("Failed to handle "+
-					"funding confirmation: %v", err)
-				return
-			}
-		}(channel)
+		err := f.sendFundingLocked(peer, channel, lnChannel,
+			shortChanID)
+		if err != nil {
+			return fmt.Errorf("failed sending fundingLocked: %v",
+				err)
+		}
+
+		fndgLog.Debugf("Channel(%v) with ShortChanID %v: successfully "+
+			"sent FundingLocked.", chanID, shortChanID)
+		return nil
 
 	// fundingLocked was sent to peer, but the channel was not added to the
 	// router graph and the channel announcement was not sent.
 	case fundingLockedSent:
-		f.wg.Add(1)
-		go func(dbChan *channeldb.OpenChannel) {
-			defer f.wg.Done()
+		err := f.addToRouterGraph(channel, shortChanID)
+		if err != nil {
+			return fmt.Errorf("failed adding to "+
+				"router graph: %v", err)
+		}
 
-			err = f.addToRouterGraph(dbChan, shortChanID)
-			if err != nil {
-				fndgLog.Errorf("failed adding to "+
-					"router graph: %v", err)
-				return
-			}
+		fndgLog.Debugf("Channel(%v) with ShortChanID %v: successfully "+
+			"added to router graph.", chanID, shortChanID)
 
-			// TODO(halseth): should create a state machine
-			// that can more easily be resumed from
-			// different states, to avoid this code
-			// duplication.
-			err = f.annAfterSixConfs(dbChan, shortChanID)
-			if err != nil {
-				fndgLog.Errorf("error sending channel "+
-					"announcements: %v", err)
-				return
-			}
-		}(channel)
+		return nil
```

Comment:
Why was the prior call to `annAfterSixConf` removed here?

Reply:
The state machine will advance to the next state (`addedToRouterGraph`) where it will be called.,